File: src/apple/Packages/MapleUI/Sources/MapleUI/Organisms/MuiEnrichmentPanel.swift
// MuiEnrichmentPanel.swift — Maple UI Organisms · Inspectors & panels
// (unified-component-catalog.md §4.3). AI-derived fields with live
// status, built from Description Field, Faces Row, Place Row, Transcript
// Block, Vision Row, Badge. No top-level loading/empty state of its own —
// each field molecule already degrades to its own placeholder/empty
// appearance given empty inputs, so this panel just wires status through
// to a small badge next to the description.

import SwiftUI

public enum MuiEnrichmentDescriptionStatus: Sendable {
    case idle, generating, done, error
}

public struct MuiEnrichmentPanel: View {
    @Binding public var description: String
    public let descriptionStatus: MuiEnrichmentDescriptionStatus
    public let people: [MuiChip]
    public let peopleRedetecting: Bool
    @Binding public var place: String
    public let placeOverridden: Bool
    public let transcriptBase: Date?
    public let transcriptEntries: [MuiTranscriptEntry]
    public let visionLabels: [MuiChip]
    @Binding public var selectedPersonId: String?
    public let descriptionRegenerate: (() -> Void)?
    public let descriptionCommitted: ((String) -> Void)?
    public let peopleRedetect: (() -> Void)?
    public let placeCommitted: ((String) -> Void)?
    public let placeCleared: (() -> Void)?

    public init(
        description: Binding<String>,
        descriptionStatus: MuiEnrichmentDescriptionStatus = .idle,
        people: [MuiChip],
        peopleRedetecting: Bool = false,
        place: Binding<String>,
        placeOverridden: Bool = false,
        transcriptBase: Date? = nil,
        transcriptEntries: [MuiTranscriptEntry] = [],
        visionLabels: [MuiChip],
        selectedPersonId: Binding<String?> = .constant(nil),
        descriptionRegenerate: (() -> Void)? = nil,
        descriptionCommitted: ((String) -> Void)? = nil,
        peopleRedetect: (() -> Void)? = nil,
        placeCommitted: ((String) -> Void)? = nil,
        placeCleared: (() -> Void)? = nil
    ) {
        self._description = description
        self.descriptionStatus = descriptionStatus
        self.people = people
        self.peopleRedetecting = peopleRedetecting
        self._place = place
        self.placeOverridden = placeOverridden
        self.transcriptBase = transcriptBase
        self.transcriptEntries = transcriptEntries
        self.visionLabels = visionLabels
        self._selectedPersonId = selectedPersonId
        self.descriptionRegenerate = descriptionRegenerate
        self.descriptionCommitted = descriptionCommitted
        self.peopleRedetect = peopleRedetect
        self.placeCommitted = placeCommitted
        self.placeCleared = placeCleared
    }

    public var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: MuiTokens.spacingMd) {
                VStack(alignment: .leading, spacing: MuiTokens.spacingXs) {
                    HStack {
                        MuiText("Description", variant: .eyebrow, color: .muted)
                        if let label = Self.statusLabel(descriptionStatus) {
                            MuiBadge(variant: .signal, value: label)
                        }
                    }
                    MuiDescriptionField(value: $description, regenerating: descriptionStatus == .generating, regenerate: descriptionRegenerate, committed: descriptionCommitted)
                }

                VStack(alignment: .leading, spacing: MuiTokens.spacingXs) {
                    MuiText("People", variant: .eyebrow, color: .muted)
                    MuiFacesRow(people: people, selectedId: $selectedPersonId, redetecting: peopleRedetecting, redetect: peopleRedetect)
                }

                VStack(alignment: .leading, spacing: MuiTokens.spacingXs) {
                    MuiText("Place", variant: .eyebrow, color: .muted)
                    MuiPlaceRow(place: $place, overridden: placeOverridden, committed: placeCommitted, cleared: placeCleared)
                }

                if let transcriptBase, !transcriptEntries.isEmpty {
                    VStack(alignment: .leading, spacing: MuiTokens.spacingXs) {
                        MuiText("Transcript", variant: .eyebrow, color: .muted)
                        MuiTranscriptBlock(baseTime: transcriptBase, entries: transcriptEntries)
                    }
                }

                VStack(alignment: .leading, spacing: MuiTokens.spacingXs) {
                    MuiText("Detected", variant: .eyebrow, color: .muted)
                    MuiVisionRow(labels: visionLabels)
                }
            }
            .padding(MuiTokens.spacingMd)
        }
    }

    // MARK: - Pure logic (unit-testable without a live view)

    /// The status badge label for the description field — `nil` (no
    /// badge) while idle, matching the web reference's `STATUS_LABEL` map.
    public static func statusLabel(_ status: MuiEnrichmentDescriptionStatus) -> String? {
        switch status {
        case .idle: return nil
        case .generating: return "Generating…"
        case .done: return "Done"
        case .error: return "Error"
        }
    }
}

#Preview("MuiEnrichmentPanel") {
    struct Demo: View {
        @State private var description = "A lone hiker crosses a black-sand beach at dusk."
        @State private var place = "Reynisfjara, Iceland"
        @State private var selectedPerson: String? = nil
        var body: some View {
            MuiEnrichmentPanel(
                description: $description,
                descriptionStatus: .done,
                people: [MuiChip(id: "1", label: "Alex")],
                place: $place,
                placeOverridden: true,
                transcriptBase: Date(),
                transcriptEntries: [MuiTranscriptEntry(id: "1", offsetMs: 0, speaker: "Alex", text: "Watch your footing here.")],
                visionLabels: [MuiChip(id: "1", label: "Beach"), MuiChip(id: "2", label: "Person")],
                selectedPersonId: $selectedPerson
            )
            .frame(width: 300, height: 520)
            .background(MuiTokens.bg)
        }
    }
    return Demo()
}

File: src/windows/Maple.WinUI/MapleUI/Organisms/MuiEnrichmentPanel.cs
using System;
using System.Collections.Generic;
using Microsoft.UI.Xaml;
using Microsoft.UI.Xaml.Controls;
using Maple.UI.Atoms;

namespace Maple.UI
{
    /// <summary>
    /// Maple.UI Enrichment Panel organism (unified-component-catalog.md
    /// §4.3, "Enrichment Panel" row: "AI-derived fields with live
    /// status", built from Description Field, Faces Row, Place Row,
    /// Transcript Block, Vision Row, Badge) — every AI-derived field for
    /// one asset, each carrying its own regenerate/redetect affordance
    /// and a status <see cref="MuiBadge"/> (e.g. "queued", "model:
    /// qwen2.5-vl") next to the section label.
    /// </summary>
    public sealed class MuiEnrichmentPanel : ContentControl
    {
        public static readonly DependencyProperty DescriptionProperty =
            DependencyProperty.Register(nameof(Description), typeof(string), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(string.Empty, (d, e) => ((MuiEnrichmentPanel)d)._description.Value = (string)e.NewValue));

        public static readonly DependencyProperty DescriptionRegeneratingProperty =
            DependencyProperty.Register(nameof(DescriptionRegenerating), typeof(bool), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(false, (d, e) => ((MuiEnrichmentPanel)d)._description.Regenerating = (bool)e.NewValue));

        public static readonly DependencyProperty PeopleProperty =
            DependencyProperty.Register(nameof(People), typeof(IReadOnlyList<MuiChip>), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(null, (d, e) => ((MuiEnrichmentPanel)d)._faces.People = (IReadOnlyList<MuiChip>?)e.NewValue));

        public static readonly DependencyProperty PlaceProperty =
            DependencyProperty.Register(nameof(Place), typeof(string), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(string.Empty, (d, e) => ((MuiEnrichmentPanel)d)._place.Place = (string)e.NewValue));

        public static readonly DependencyProperty TranscriptEntriesProperty =
            DependencyProperty.Register(nameof(TranscriptEntries), typeof(IReadOnlyList<MuiTranscriptEntry>), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(null, (d, e) => ((MuiEnrichmentPanel)d)._transcript.Entries = (IReadOnlyList<MuiTranscriptEntry>?)e.NewValue));

        public static readonly DependencyProperty VisionLabelsProperty =
            DependencyProperty.Register(nameof(VisionLabels), typeof(IReadOnlyList<MuiChip>), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(null, (d, e) => ((MuiEnrichmentPanel)d)._vision.Labels = (IReadOnlyList<MuiChip>?)e.NewValue));

        public static readonly DependencyProperty StatusLabelProperty =
            DependencyProperty.Register(nameof(StatusLabel), typeof(string), typeof(MuiEnrichmentPanel),
                new PropertyMetadata(null, (d, _) => ((MuiEnrichmentPanel)d).Rebuild()));

        public string Description { get => (string)GetValue(DescriptionProperty); set => SetValue(DescriptionProperty, value); }
        public bool DescriptionRegenerating { get => (bool)GetValue(DescriptionRegeneratingProperty); set => SetValue(DescriptionRegeneratingProperty, value); }
        public IReadOnlyList<MuiChip>? People { get => (IReadOnlyList<MuiChip>?)GetValue(PeopleProperty); set => SetValue(PeopleProperty, value); }
        public string Place { get => (string)GetValue(PlaceProperty); set => SetValue(PlaceProperty, value); }
        public IReadOnlyList<MuiTranscriptEntry>? TranscriptEntries { get => (IReadOnlyList<MuiTranscriptEntry>?)GetValue(TranscriptEntriesProperty); set => SetValue(TranscriptEntriesProperty, value); }
        public IReadOnlyList<MuiChip>? VisionLabels { get => (IReadOnlyList<MuiChip>?)GetValue(VisionLabelsProperty); set => SetValue(VisionLabelsProperty, value); }
        public string? StatusLabel { get => (string?)GetValue(StatusLabelProperty); set => SetValue(StatusLabelProperty, value); }

        public event EventHandler? DescriptionRegenerateRequested;
        public event EventHandler<string>? DescriptionCommitted;
        public event EventHandler? FacesRedetectRequested;
        public event EventHandler<string>? PlaceCommitted;
        public event EventHandler? PlaceCleared;

        private readonly StackPanel _root = new() { Orientation = Orientation.Vertical, Spacing = 20 };
        private readonly MuiBadge _statusBadge = new() { Variant = MuiBadgeVariant.Signal };
        private readonly MuiDescriptionField _description = new() { Placeholder = "No description yet" };
        private readonly MuiFacesRow _faces = new();
        private readonly MuiPlaceRow _place = new();
        private readonly MuiTranscriptBlock _transcript = new();
        private readonly MuiVisionRow _vision = new();

        public MuiEnrichmentPanel()
        {
            _root.Children.Add(Section("Description", _description));
            _root.Children.Add(Section("People", _faces));
            _root.Children.Add(Section("Place", _place));
            _root.Children.Add(Section("Transcript", _transcript));
            _root.Children.Add(Section("Vision labels", _vision));
            Content = _root;

            _description.Regenerate += (_, _) => DescriptionRegenerateRequested?.Invoke(this, EventArgs.Empty);
            _description.Committed += (_, text) => { Description = text; DescriptionCommitted?.Invoke(this, text); };
            _faces.Redetect += (_, _) => FacesRedetectRequested?.Invoke(this, EventArgs.Empty);
            _place.Committed += (_, text) => { Place = text; PlaceCommitted?.Invoke(this, text); };
            _place.Cleared += (_, _) => PlaceCleared?.Invoke(this, EventArgs.Empty);

            Rebuild();
        }

        private UIElement Section(string label, UIElement body)
        {
            var stack = new StackPanel { Orientation = Orientation.Vertical, Spacing = 8 };
            var heading = new StackPanel { Orientation = Orientation.Horizontal, Spacing = 8 };
            heading.Children.Add(new MuiText { Text = label, Variant = MuiTextVariant.Eyebrow, ColorRole = MuiTextColorRole.Muted });
            if (label == "Description") heading.Children.Add(_statusBadge);
            stack.Children.Add(heading);
            stack.Children.Add(body);
            return stack;
        }

        private void Rebuild()
        {
            _statusBadge.Visibility = string.IsNullOrEmpty(StatusLabel) ? Visibility.Collapsed : Visibility.Visible;
            _statusBadge.Value = StatusLabel ?? string.Empty;
        }
    }
}
